aim/web/ui_v2/src/utils/d3/drawHoverAttributes.ts

```diff
@@ -68,85 +63,133 @@ function drawHoverAttributes(props: IDrawHoverAttributesProps): void {
+          .attr('y2', (axisLine: IAxisLineData) => axisLine.y2)
+          .lower();
+
+        // Draw Circles
+        attributesNodeRef.current
+          .selectAll('circle')
+          .data(nearestCircles)
+          .join('circle')
+          .attr('class', 'HoverCircle')
+          .attr('id', (circle: INearestCircle) => `Circle-${circle.key}`)
+          .attr('clip-path', 'url(#circles-rect-clip-' + index + ')')
+          .attr('cx', (circle: INearestCircle) => circle.x)
+          .attr('cy', (circle: INearestCircle) => circle.y)
+          .attr('r', CircleEnum.Radius)
+          .style('fill', (circle: INearestCircle) => circle.color)
+          .on('click', function (this: SVGElement, circle: INearestCircle) {
+            // TODO handle click
+            d3.select(this).classed('active', false).classed('focus', true);
+          });
+
+        // set active circle
+        d3.select(`[id=Circle-${closestCircle.key}]`)
+          .attr('r', CircleEnum.ActiveRadius)
+          .classed('active', true)
+          .raise();
+
+        setAxisLabel({
+          closestCircle,
+          visAreaRef,
+          visBoxRef,
+          plotBoxRef,
+          xAxisLabelNodeRef,
+          yAxisLabelNodeRef,
+          xScale: attributesRef.current.xScale,
+          yScale: attributesRef.current.yScale,
+          xAlignment,
+        });
+      }
 
-    const axisLineData: IAxisLineData[] = [
-      { x1: closestCircle.x, y1: 0, x2: closestCircle.x, y2: height },
-      { x1: 0, y1: closestCircle.y, x2: width, y2: closestCircle.y },
-    ];
-
-    // Draw horizontal/vertical lines
-    attributesNodeRef.current
-      .selectAll('line')
-      .data(axisLineData)
-      .join('line')
-      .attr('class', `${classes.HoverLine}`)
-      .style('stroke-width', 1)
-      .style('stroke-dasharray', '4 2')
-      .style('fill', 'none')
-      .lower()
-      .attr('x1', (axisLine: IAxisLineData) => axisLine.x1)
-      .attr('y1', (axisLine: IAxisLineData) => axisLine.y1)
-      .attr('x2', (axisLine: IAxisLineData) => axisLine.x2)
-      .attr('y2', (axisLine: IAxisLineData) => axisLine.y2);
+      closestCircleRef.current = closestCircle;
+    }
   }
 
   attributesRef.current.updateHoverAttributes = handleMouseMove;
-  svgArea?.on('mousemove', handleMouseMove);
 
-  svgArea?.on('mouseleave', (event) => {
-    // TODO handle mouseLeave
+  svgArea?.on('mousemove', (event: MouseEvent) => {
+    handleMouseMove(d3.pointer(event));
+  });
+
+  svgArea?.on('mouseleave', () => {
+    if (closestCircleRef.current?.key) {
+      linesNodeRef.current
+        .select(`[id=Line-${closestCircleRef.current.key}]`)
+        .classed('active', false);
+
+      attributesNodeRef.current
+        .select(`[id=Circle-${closestCircleRef.current.key}]`)
+        .attr('r', CircleEnum.Radius)
+        .classed('active', false);
+
+      attributesNodeRef.current.select('.HoverLine-y').remove();
+
+      yAxisLabelNodeRef.current?.remove();
```

Comment:
I guess here we should reset opacity of not highlight lines as well

Reply:
You mean on mouse leave we should reset highlight mode to highlight-off mode ?

Reply:
No, we just need to reset the highlighted line state, so no line differs from the others.

Reply:
ok, i will fix this